online-shop/src/app/product-detail/product-detail.component.html

```diff
@@ -0,0 +1,33 @@
+<div *ngIf="product">
+  <h2>{{product.name | uppercase}} Details</h2>
+  <div><span>id: </span>{{product.id}} <br>
+       <span>Name: </span>{{product.name}} <br>
+       <span>Category: </span>{{product.category}} <br>
+       <span>Price: </span>{{product.price}} <br>
+       <span>Image: </span>{{product.image}} <br>
+       <span>Description: </span>{{product.description}} <br>
+
+  <br>
+  <button class="button1" (click)="addToCart()">Add to cart</button><br><br>
```

Comment:
action should be available only for customers. you can add an ng-if and hide the button for non customer roles